src/SdkGenerator/Templates/dart/ApiClient.dart.scriban

```diff
@@ -2,193 +2,168 @@ import 'dart:async';
 import 'dart:convert';
 import 'dart:io';
 
-import 'package:flutter/foundation.dart';
-import 'package:flutter/widgets.dart';
-import 'package:flutter_toolkit/http/APIClient.dart';
-import 'package:http/http.dart' as http;
 import 'package:brotli/brotli.dart';
+import 'package:flutter/foundation.dart';
+import 'package:http/http.dart';
+
+import '../client/AuthService.dart';
+import '../flavours/BuildConfig.dart';
 
 {{~ for cat in api.categories ~}}
-import 'clients/{{ cat }}Client.dart';
+import './clients/{{ cat | camel_case_to_snake_case }}_client.dart';
 {{~ end ~}}
-import '{{ project.dart.response_class }}.dart';
-import 'AstroError.dart';
-import '../flavours/BuildConfig.dart';
-import '../client/AuthService.dart';
-import '{{ project.dart.class_name }}.dart';
 
+import 'astro_error.dart';
+import '{{ project.dart.response_class | camel_case_to_snake_case }}.dart';
+import '{{ project.dart.class_name | camel_case_to_snake_case }}.dart';
 
 class {{ project.dart.class_name }}Impl implements {{ project.dart.class_name }} {
-  BuildConfig config;
-
-  Map<String, String>? headers;
+  {{ project.dart.class_name }}Impl(this.authService, this._client, this.apiUrl);
 
-  AuthService authService;
-  APIClient api;
-
-  {{ project.dart.class_name }}Impl(this.authService, this.api, this.config);
-  
+  final BaseClient _client;
+  final String apiUrl;
+  final AuthService authService;
 {{~ for cat in api.categories ~}}
 
   {{ cat }}Client? _{{ cat | to_camel_case }}Client;
-
   @override
-  {{ cat | to_camel_case }}() {
+  {{ cat }}Client {{ cat | to_camel_case }}() {
     return _{{ cat | to_camel_case }}Client ??= {{ cat }}Client(this);
   }
 {{~ end ~}}
 
-  ClientResponse _logRequest(
-      String action, String path, int startMillis, value) {
+  Response _logRequest(String action, String path, int startMillis, Response value) {
     if (!BuildConfig.isRelease) {
-      debugPrint(
-          "$action $path took ${DateTime.now().millisecondsSinceEpoch - startMillis}ms");
+      debugPrint('$action $path took ${DateTime.now().millisecondsSinceEpoch - startMillis}ms');
     }
-    //Todo other stuff
+
     return value;
   }
 
-  {{ project.dart.response_class }}<dynamic> _handleAPIError(error) {
-    debugPrint("API Error $error");
+  {{ project.dart.response_class }}<T> _handleAPIError<T>(dynamic error) {
```

Comment:
Future idea: Suggest we move this method entirely into the response class, instead of leaving it partially here.